### PR TITLE
[ETHEREUM-CONTRACTS] full API in ISuperfluid

### PR DESCRIPTION
**Pull request by d10r**
Rationale:
`ISuperfluid` used to provide an almost(*) complete set of Solidity APIs for the Superfluid protocol.
Recently added interfaces are missing and were added in this PR.

(*) while this did include ERC20 and ERC777 APIs, it did not include the agreement APIs for CFA and IDA. Those were also added in this PR.

### CODE REVIEW

**Comment by hellwolf** [comment]
maybe piggyback this change here (to address a warning, v2 -> v3):

```
nix-shell$ git grep actions/checkout@v2 ../../
../../.github/workflows/call.test-ethereum-contracts.yml:      uses: actions/checkout@v2
```

We will discuss about how to manage this umbrella import thing next time.

File: .github/workflows/call.test-ethereum-contracts.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
 
     steps:
     - name: Check out code into the Go module directory
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
```
Comment (hellwolf) [inline .github/workflows/call.test-ethereum-contracts.yml:23]: thanks!

**Comment by hellwolf** [comment]
My systematic search for the offenders (importing stuff from non ISuperfluid.sol):

```
$ git grep -Eo "\".*/ethereum-contracts/contracts/interfaces/.*\"" | grep -v ISuperfluid.sol
packages/automation-contracts/autowrap/contracts/Manager.sol:"../../../ethereum-contracts/contracts/interfaces/agreements/IConstantFlowAgreementV1.sol"
packages/automation-contracts/autowrap/test/Manager.t.sol:"../../../ethereum-contracts/contracts/interfaces/tokens/ISETH.sol"
packages/automation-contracts/autowrap/test/WrapStrategy.t.sol:"../../../ethereum-contracts/contracts/interfaces/tokens/ISETH.sol"
packages/automation-contracts/autowrap/test/WrapUp.t.sol:"../../../ethereum-contracts/contracts/interfaces/tokens/ISETH.sol"
packages/automation-contracts/scheduler/test/FlowScheduler.t.sol:"../../../ethereum-contracts/contracts/interfaces/superfluid/ISuperToken.sol"
packages/automation-contracts/scheduler/test/FlowSchedulerResolver.t.sol:"../../../ethereum-contracts/contracts/interfaces/superfluid/ISuperToken.sol"
packages/automation-contracts/scheduler/test/VestingScheduler.t.sol:"../../../ethereum-contracts/contracts/interfaces/superfluid/ISuperToken.sol"
packages/ethereum-contracts/README.md:"@superfluid-finance/ethereum-contracts/contracts/interfaces/agreements/IConstantFlowAgreementV1.sol"
```

ISETH.sol and ISuperToken.sol probably okay. Others (one is readme):

```
packages/automation-contracts/autowrap/contracts/Manager.sol:"../../../ethereum-contracts/contracts/interfaces/agreements/IConstantFlowAgreementV1.sol"
packages/ethereum-contracts/README.md:"@superfluid-finance/ethereum-contracts/contracts/interfaces/agreements/IConstantFlowAgreementV1.sol"
```

File: packages/ethereum-contracts/contracts/interfaces/agreements/IInstantDistributionAgreementV1.sol [hunk 2]
```diff
@@ -4,7 +4,6 @@ pragma solidity >= 0.8.11;
 import { ISuperAgreement } from "../superfluid/ISuperAgreement.sol";
 import { ISuperfluidToken } from "../superfluid/ISuperfluidToken.sol";
 
-
```
Comment (hellwolf) [inline packages/ethereum-contracts/contracts/interfaces/agreements/IInstantDistributionAgreementV1.sol:7]: heh, it's my weird habbit of putting two liens behind import blocks.

**Review by hellwolf — COMMENTED** [review]
looks good! some small comments maybe relevant.

**Review by 0xdavinchee — APPROVED** [review]
LGTM

**Review by hellwolf — APPROVED** [review]

**Pull request merged**